app/client/components/analytics.tsx

```diff
@@ -77,85 +78,102 @@ export const applyAnyOptimiseExperiments = () => {
   }
 };
 
-export class AnalyticsTracker extends React.PureComponent<{}> {
-  public readonly INTCMP?: string;
+export const AnalyticsTracker = () => {
+  const [cmpIsInitialised, setCmpIsInitialised] = useState<boolean>(false);
 
-  constructor(props: {}) {
-    super(props);
-    if (typeof window !== "undefined" && window.ga) {
-      const queryParams = parse(window.location.href, true).query;
+  if (typeof window !== "undefined" && window.ga) {
+    const queryParams = parse(window.location.href, true).query;
 
-      this.INTCMP = queryParams.INTCMP;
+    const INTCMP = queryParams.INTCMP;
 
-      if (window.guardian) {
-        // tslint:disable-next-line:no-object-mutation
-        window.guardian.INTCMP = this.INTCMP;
+    if (window.guardian) {
+      // tslint:disable-next-line:no-object-mutation
+      window.guardian.INTCMP = INTCMP;
 
-        const abName = queryParams.abName;
-        const abVariant = queryParams.abVariant;
+      const abName = queryParams.abName;
+      const abVariant = queryParams.abVariant;
 
-        if (abName && abVariant) {
-          window.guardian.abTest = {
-            name: abName,
-            variant: abVariant
-          };
-        }
+      if (abName && abVariant) {
+        // tslint:disable-next-line:no-object-mutation
+        window.guardian.abTest = {
+          name: abName,
+          variant: abVariant
+        };
       }
```

Comment:
Why is this code needed?

Reply:
it comes from https://github.com/guardian/manage-frontend/pull/197 (and identity thing)